### PR TITLE
feat: added default language properly in core

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Modified the default language handling

### CODE REVIEW

File: Editor/ProjectSettings/TextToSpeechSectionProvider.cs
Comment (Gusinuhe) [inline Editor/ProjectSettings/TextToSpeechSectionProvider.cs:9]: Missing docs all over the class

**Review by Gusinuhe — APPROVED** [review]

**Review by Gusinuhe — APPROVED** [review]

**Comment by Luphoxa** [comment]
Moved here from the Core PR: 
Test Notes:
👍 Language put to EN or DE shows the corresponding hover text in EN or DE  
👍  Changing to a language that has no desktop mode txt file, the default desktop text is shown 
👎 Creating a language file for a new language (I copied and renamed the EN file and changed the text) and setting this language in the language settings does change it correctly for TTS, but the desktop mode still uses the default txt file, although a file for it exists. Maybe I forgot to add something? Need to have a look here, maybe together if you want  

Will add more results if I find something else :)

**Review by tomwim — APPROVED** [review]

**Comment by tomwim** [comment]
Blocked until https://github.com/Innoactive/Creator/pull/266 is merged

**Comment by Luphoxa** [comment]
This still happens to me
> 👎 Creating a language file for a new language (I copied and renamed the EN file and changed the text) and setting this language in the language settings does change it correctly for TTS, but the desktop mode still uses the default txt file, although a file for it exists. Maybe I forgot to add something? Need to have a look here, maybe together if you want
> 
Is this related to this branch or to https://github.com/Innoactive/Creator/pull/266 ?
Or did I name my ES file wrong? (`Creator_ES_desktop_mode`)

But I found nothing else.

**Comment by SimonTheSourcerer** [comment]
Do not use the resource folders, please use the localization folder in streaming assets which would require you to create a ES_desktop_mode.json

**Comment by SimonTheSourcerer** [comment]
otherwise if you try out different files setup please provide a picture of the setup otherwise i can only guess.

**Comment by tomwim** [comment]
@Luphoxa does this mean you're approving this PR?

**Comment by Luphoxa** [comment]
My setup is this, like you thought. But why does my EN and DE file work from here but additional languages need to go into the streaming assets?
image

I also created a Localization folder in StreamingAssets in the Training and it did not work
image

Then I moved it directly to the StreamingAssets and it did not work either.
image

I think our main problem is that the usability of the localization is not that great and not explained anywhere. So I don't know if I did this right or not. And how "right" would look like.

**Comment by SimonTheSourcerer** [comment]
here is the new documentation which will be published with the changes on the localization system: 
https://github.com/Innoactive/Creator/blob/develop/.Documentation/articles/innoactive-creator/localization.md

**Review by Gusinuhe — APPROVED** [review]

**Comment by Luphoxa** [comment]
Okay so I tried these paths:
For these, I copied the file from resources `Creator_EN_desktop_mode` and changed its name and content to es. 
I created a Trainings Folder, a Folder with the Course name, and a Localization folder for this: (this is the one suggested in the docu)
`Assets/StreamingAssets/Trainings/{Course Name}/Localization`
I created a Localization folder for these:
`Assets/StreamingAssets/Training/{Course Name}/Localization`
`Assets/StreamingAssets/Training/Localization`
`Assets/StreamingAssets/Localization`

-> I always get the default english.

In the documentation, the files are of tye .json so I tried it as json in these:
`Assets/StreamingAssets/Training/{Course Name}/Localization` -> default
`Assets/StreamingAssets/Trainings/{Course Name}/Localization` -> default

Then I additionally added an ES.json, maybe it's needed:
`Assets/StreamingAssets/Training/{Course Name}/Localization` -> default
`Assets/StreamingAssets/Trainings/{Course Name}/Localization` -> default

Defining default:
- I get the spanish Audio, but the english hover text from the default engish desktop file, not the EN file

Innoactive Language Settings:
image

**Review by Luphoxa — CHANGES_REQUESTED** [review]
I would like to be sure that additional languages apart from en and de are possible before merging.
Or if this is not part of this branch directly, but from another branch, dismiss this review here, but make sure to note down this bug.

**Comment by SimonTheSourcerer** [comment]
I added this to my course/Localization:
image
and I have spanish text displayed and also spanish voice. Works perfectly as it should be only thing i found, i cant

**Review by Luphoxa — APPROVED** [review]

**Comment by Luphoxa** [comment]
The issue was that the file I copied had a `, ` at the bottom which lead to a json parser issue for my file, where I copied and adapted the text. Simon has kindly agreed to change this in the CreatorPro PR.
Now it works 👍  Thanks for your patience!

**Pull request merged**
